src/include/distributed/resource_lock.h

```diff
@@ -169,7 +169,7 @@ IsNodeWideObjectClass(ObjectClass objectClass)
 	 * If new object classes are added and none of them are node-wide, then update
 	 * this assertion check based on latest supported major Postgres version.
 	 */
-	StaticAssertStmt(PG_MAJORVERSION_NUM <= 17,
```

Comment:
Did you check if there are any newly added ObjectClass'es that are node-wide in PG18?

Reply:
It seems that no new global catalogs appeared in PG 18.

Since `ObjectClass` no longer exists in PG 18, Should we drop or narrow to 16 that `StaticAssertStmt` rather than expanding it? 

By PG 18 there is no `ObjectClass` enum at all, so `StaticAssertStmt(PG_MAJORVERSION_NUM <= 18, …)` never really exercises anything useful, right?